### PR TITLE
Add Api endpoint to fetch district teams

### PR DESCRIPTION
**Pull request by cmlicata**
Added an API endpoint to fetch all DistrictTeams in a given district and year. Also added the district team converter to model_to_dict.py. Tests will be added in the next commit
Towards #1162 

From @phil-lopreiato:
 - Still needs to update apidocs and cache clearing. I can take care of those.

### CODE REVIEW

File: database/event_query.py [hunk 1]
```diff
@@ -40,6 +41,25 @@ def _query_async(self):
         raise ndb.Return(events)
 
 
+class DistrictTeamsQuery(DatabaseQuery):
+    CACHE_VERSION = 0
+    CACHE_KEY_FORMAT = 'district_events_{}'  # (district_key)
```
Comment (gregmarra) [inline database/event_query.py]: should this be `district_events`, given this is `DistrictTeamsQuery`

**Comment by gregmarra** [comment]
@cmlicata this branch has merge conflicts now.

File: database/event_query.py [hunk 2]
```diff
@@ -40,6 +41,25 @@ def _query_async(self):
         raise ndb.Return(events)
 
 
+class DistrictTeamsQuery(DatabaseQuery):
```
Comment (fangeugene) [inline database/event_query.py]: Isn't this already here? 
https://github.com/the-blue-alliance/the-blue-alliance/blob/master/database/team_query.py#L52

File: database/event_query.py [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 from consts.district_type import DistrictType
 from database.database_query import DatabaseQuery
+from models.district_team import DistrictTeam
```
Comment (fangeugene) [inline database/event_query.py]: Unused import?

Comment (cmlicata) [inline database/event_query.py]: yea, sorry about that.  Just fixed it.  Thanks for the heads up!

File: helpers/model_to_dict.py [hunk 4]
```diff
@@ -141,3 +142,17 @@ def robotConverter(self, robot):
         robot_dict["year"] = robot.year
         robot_dict["name"] = robot.robot_name
         return robot_dict
+
+    @classmethod
+    def districtTeamConverter(self, team):
```
Comment (fangeugene) [inline helpers/model_to_dict.py]: I don't believe this is used.

File: api_main.py [hunk 5]
```diff
@@ -93,12 +94,15 @@
                                webapp2.Route(r'/api/v2/district/<district_abbrev:>/<year:([0-9]*)>/events',
                                              ApiDistrictEventsController,
                                              methods=['GET']),
+                               webapp2.Route(r'/api/v2/district/<district_abbrev:>/<year:([0-9]*)>/teams',
+                                             ApiDistrictTeamsController,
+                                             methods=['GET']),
```
Comment (phil-lopreiato) [inline api_main.py]: @cmlicata You're going to have to add the `OPTIONS` method here. See #1314

**Comment by cmlicata** [comment]
@gregmarra @fangeugene Resolved the conflict and this should be good to go. Sorry about that.

File: api_main.py [hunk 6]
```diff
@@ -4,7 +4,8 @@
 import tba_config
 
 from controllers.api_controller import ApiDeprecatedController, CsvTeamsAll
-from controllers.api.api_district_controller import ApiDistrictListController, ApiDistrictEventsController, ApiDistrictRankingsController
+from controllers.api.api_district_controller import ApiDistrictListController, ApiDistrictTeamsController, ApiDistrictRankingsController, \
+    ApiDistrictEventsController
```
Comment (gregmarra) [inline api_main.py]: We should make this indentation consistent w below (and other files)

**Comment by gregmarra** [comment]
@fangeugene  you want to check this out? I can look more in depth later.

**Comment by fangeugene** [comment]
Yep I'll take a detailed look tonight.

**Comment by fangeugene** [comment]
Looks good.

**Pull request merged**